multicluster/controllers/multicluster/serviceexport_controller.go

```diff
@@ -338,6 +331,26 @@ func (r *ServiceExportReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	return ctrl.Result{}, nil
 }
 
+func (r *ServiceExportReconciler) checkRemoteManager() (bool, commonarea.RemoteCommonArea, error) {
+	defer r.mutex.Unlock()
+	r.mutex.Lock()
+	if *r.remoteCommonAreaManager == nil {
+		klog.InfoS("ClusterSet has not been initialized properly, no remote cluster manager")
+		return true, nil, nil
+	}
+	r.localClusterID = string((*r.remoteCommonAreaManager).GetLocalClusterID())
+	if len(r.localClusterID) == 0 {
+		klog.InfoS("localClusterID is not initialized, skip reconciling")
+		return true, nil, nil
+	}
+
+	remoteCluster, err := getRemoteCommonArea(r.remoteCommonAreaManager)
+	if err != nil {
+		return false, nil, err
+	}
+	return false, remoteCluster, nil
```

Comment:
We never access any mutable attribute of remoteCluster after releasing the lock right?

Reply:
yes, once leader and member cluster is setup, the cluster ID and namespace are not changed, and following code are using common area's client to get cache data (thread safe) or connect to leader API directly.